uvtools/dspec.py

```diff
@@ -226,34 +581,46 @@ def high_pass_fourier_filter(data, wgts, filter_size, real_delta, clean2d=False,
+                    elif mode == 'dayenu':
+                        d_r, info_here = dayenu_filter(np.arange(len(data[i]))*real_delta,
+                                                       data[i] * wgts[i] * win, wgts[i] * win,
+                                                       filter_dimensions=[1], filter_centers=fc,
+                                                       filter_half_widths=fw, filter_factors=ff, cache=cache)
                         _d_res[i] = np.fft.ifft(d_r)
-                        if deconv_linear_foregrounds:
+                        if deconv_dayenu_foregrounds:
                             if fg_deconv_method == 'clean':
                                 _d_cl[i], info_fg = aipy.deconv.clean(_d[i] - _d_res[i], _w[i], area=area_fg, tol=tol, stop_if_div=False, maxiter=maxiter, gain=gain)
                                 del info_fg['res']
                                 info_fg['maxiter'] = maxiter
                                 info_fg['gain'] = gain
                                 info_here['fg_deconv'] = info_fg
                             elif fg_deconv_method == 'leastsq':
-                                nmin = int((fcfg[0] - fwfg[0]) * real_delta * data.shape[-1])
-                                nmax = int((fcfg[0] + fwfg[0]) * real_delta * data.shape[-1])
+                                nmin = int((fcfg[0] - fwfg[0]) * real_delta * fg_deconv_fundamental_period[-1])
+                                nmax = int((fcfg[0] + fwfg[0]) * real_delta * fg_deconv_fundamental_period[-1])
                                 info_here['fg_deconv'] = {'method':'leastsq','nmin':nmin, 'nmax':nmax}
                                 d_cl, _, _ = delay_filter_leastsq_1d( (data[i] * wgts[i] * win - d_r).squeeze(), flags=(wgts[i]==0.).squeeze(), sigma=1.,
-                                                                    nmax=(nmin, nmax), freq_units=True, even_modes=True)
+                                                                    nmax=(nmin, nmax), freq_units=True, even_modes=True, fundamental_period=fg_deconv_fundamental_period[-1])
                                 _d_cl[i] = np.fft.ifft(d_cl)
                         else:
                             _d_cl[i] = _d[i] - _d_res[i]
                         info.append(info_here)
 
+                    elif mode == 'dft_interp':
+                        info_here = {}
+                        nmin = int((fcfg[0] - fwfg[0]) * real_delta * fg_deconv_fundamental_period[-1])
+                        nmax = int((fcfg[0] + fwfg[0]) * real_delta * fg_deconv_fundamental_period[-1])
+                        info_here['fg_deconv'] = {'method':'dft_interp','nmin':nmin, 'nmax':nmax}
+                        d_cl, _, _ = delay_filter_leastsq_1d( (data[i] * wgts[i] * win ).squeeze(), flags=(wgts[i]==0.).squeeze(), sigma=1.,
```

Comment:
the conversion to flags is better done as
`flags=np.isclose(wgts[i], 0).squeeze()`
because we should avoid float-to-float equivalence

Reply:
i've changed this.